tests/test_frontend_type_conversion.py

```diff
@@ -56,7 +56,7 @@ def to_xdsl() -> Callable[..., Any]:
 
 
 def test_raises_exception_on_unknown_type():
-    type_converter = TypeConverter(globals)
+    type_converter = TypeConverter(globals, 'file')
```

Comment:
Why not defaulting to None though? Otherwise error messages get inconsistent and it seems weird passing this string around in tests?

Reply:
It's just that bits of code assume it's not `None`. I can have a go at making it all optional and seeing where that goes

- [ ] make file optional and see where that breaks

Reply:
Yeah it's also ok being optional everywhere